src/code/z_play.c

```diff
@@ -1,5 +1,25 @@
-#include "global.h"
+#include "z64.h"
+#include "regs.h"
+#include "functions.h"
+#include "z64vismono.h"
+
+// Variables are put before most headers as a hacky way to bypass bss reordering
```

Comment:
monkaBss. Is this stable as `z64.h` and `functions.h` changes over time?

Reply:
Yes! Or at least it is for this kind of bss reordering.
The kind of bss reordering I'm trying to prevent here is the one produced by `extern`ed bss symbols from this file. This kind consists on `extern`s symbols from this file being either reordered or missing end up affecting the bss order of the variables of this file.

Concrete example, all externs:
```c
/* variables.h */

// All the bss variables of example.c are externed here, so this kind doesn't affect. Notice a few should not be global tho.
extern s32 sExampleA;
extern s32 gExampleB;
extern s32 sExampleC;

/* example.c */

#include "variables.h"

s32 sExampleA;
s32 gExampleB;
s32 sExampleC;
```

Concrete example, some externs, plus reordering:
```c
/* variables.h */

// Only the global bss variables are present in the header
extern s32 gExampleB;

/* example.c */

#include "variables.h"

// IDO has seen `gExampleB` before `sExampleA`, so it will reorder them, which is bad :c 
s32 sExampleA;
s32 gExampleB;
s32 sExampleC;
```

Concrete example, some externs, avoiding reordering:
```c
/* variables.h */

// Only the global bss variables are present in the header
extern s32 gExampleB;


/* example.c */

// IDO has not seen declarations for those symbols, so it doesn't reorder them (for this specific bss reordering kind)
s32 sExampleA;
s32 gExampleB;
s32 sExampleC;

#include "variables.h"

```

As proof for this hacky way working is this PR https://github.com/zeldaret/mm/pull/1199 not having bss issues in the two files I did this

Reply:
This is only going to work for basic types though, right? Hopefully this will evaporate once we get rid of the vast majority of variables.h.

Reply:
By not means this is a definitive solution. Hopefully we will have the headers split sooner than later

Reply:
I think I'm following now. I did the same thing for `z_demo.c` and got the following issue:

If you extern the global bss variables in `z64cutscene.h`, then this hack means that the bss variables need to be declared before including `z64cutscene.h`. But then `CutsceneCamera` is defined, meaning the variables externs have to be put to `variables.h`.

A different solution was to extern all variables, whether global or not, to preserve the order and that seemed to work for z_demo.c, which seems like a nice temporary solution until headers are more split.

Would that same trick work here in `z_play.c`?

Reply:
Oh wait I see. That's how it was before, everything was externed in `variables.h` and this trick is needed now that everything is not externed

Reply:
Yeah, so I think the question becomes which `hack` do we prefer. Only externing some and messing with include ordering, or externing all including static bss that shouldn't be accessed globally. Tbh I do not have a particular preference one way or the other.

Reply:
Oh yeah, i forgot to mention the important part, sorry.
Yes, this is needed because I removed the `extern`s that are not used outside this file. If IDO sees the `extern`s before the actual bss definitions so it gives priority to the order of those `extern`s instead of the actual symbols definitions

Reply:
Alright, I suppose this is fine. Could even forward declare these structs to have even less headers above if you want, although I'm not sure how far we want to take it